codegen/src/lib.rs

```diff
@@ -8,6 +8,20 @@ use quote::quote;
 use syn::Error;
 use syn::ItemFn;
 
+//We can't use procedural macros as expressions due to rust limitations
+//so this generates a function instead
+#[proc_macro]
+pub fn make_env(input: TokenStream) -> TokenStream {
+    let env_arg = syn::parse_macro_input!(input as syn::LitStr);
+    let app_size_str = std::env::var(env_arg.value()).unwrap();
+    let size = app_size_str.parse::<usize>().unwrap();
```

Comment:
What happens when reading or parsing the environment variable fails? I think, error should be handled by returning `syn::Error`.

Reply:
I wasn't sure how valuable it would be to try and handle all the errors vs. just letting the parse fail at the unwrap level.

Reply:
The advantage is that you get a correct error location in the error output.